src/RenderStatistic.cc

```diff
@@ -0,0 +1,98 @@
+/*
+ *  OpenSCAD (www.openscad.org)
+ *  Copyright (C) 2020 Golubev Alexander <fatzer2@gmail.com>
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  As a special exception, you have permission to link this program
+ *  with the CGAL library and distribute executables, as long as you
+ *  follow the requirements of the GNU GPL in regard to all of the
+ *  software in the executable aside from CGAL.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ */
+
+#include "printutils.h"
+#include "GeometryCache.h"
+#include "CGALCache.h"
+#include "polyset.h"
+#include "Polygon2d.h"
+
+#ifdef ENABLE_CGAL
+#include "CGAL_Nef_polyhedron.h"
+#endif // ENABLE_CGAL
+
+#include "RenderStatistic.h"
+
+
+void RenderStatistic::printCacheStatistic()
+{
+  GeometryCache::instance()->print();
+#ifdef ENABLE_CGAL
+  CGALCache::instance()->print();
+#endif
+}
+
+void RenderStatistic::printRenderingTime(std::chrono::milliseconds ms)
+{
+  PRINTB("Total rendering time: %02d:%02d.%02d",
```

Comment:
Shouldn't the last one be `%03d`?

As discussed we also should add back the hours. I don't mind either way, just as leading value like `1:20:01.238` or going back to the original format (with added milliseconds).

Reply:
thanks, good catch on the microseconds format.